### PR TITLE
K8s etcd scale policy

### PR DESCRIPTION
**Pull request by LLParse**

### CODE REVIEW

File: templates/kubernetes/2/rancher-compose.yml [hunk 1]
```diff
@@ -1,8 +1,7 @@
 .catalog:
   name: "Kubernetes"
-  version: "v1.2.4-rancher6"
+  version: "v1.2.4-rancher7"
   description: "Rancher Kubernetes service"
```
Comment (alena1108) [inline templates/kubernetes/2/rancher-compose.yml]: we should put a minimum rancher version as it's not supposed to work with pre 1.1 (no scale policy support prior to that)

File: templates/kubernetes/2/docker-compose.yml [hunk 2]
```diff
@@ -44,47 +44,30 @@ proxy:
...
   - discovery
   volumes_from:
-  - etcd-data
+    - data
 
-etcd-data:
-  image: busybox
-  command: /bin/true
+data:
+  image: llparse/etcd
```
Comment (alena1108) [inline templates/kubernetes/2/docker-compose.yml]: shall we host this image under rancher dockerhub?

Comment (LLParse) [inline templates/kubernetes/2/docker-compose.yml]: You know what? I don't see enough value in the :latest hack to warrant changing from busybox. I'm going to change it back.

I'll build the rancher/etcd:v2.3.6-2 and update this PR along with that change.

File: templates/kubernetes/2/rancher-compose.yml [hunk 3]
```diff
@@ -0,0 +1,77 @@
+.catalog:
+  name: Kubernetes
+  version: v1.2.4-rancher7
+  description: Rancher Kubernetes service
+  minimum_rancher_version: v1.1.0-rc2
```
Comment (alena1108) [inline templates/kubernetes/2/rancher-compose.yml]: it won't be an RC, but rather the official rancher version 1.1

Comment (LLParse) [inline templates/kubernetes/2/rancher-compose.yml]: d'oh

Comment (alena1108) [inline templates/kubernetes/2/rancher-compose.yml]: the official version would be v1.1.0

Comment (LLParse) [inline templates/kubernetes/2/rancher-compose.yml]: Yep, made the change, squash merged so it should show up in the commit below.

File: templates/kubernetes/config.yml [hunk 4]
```diff
@@ -1,5 +1,5 @@
 name: Kubernetes
 description: |
   Kubernetes service
-version: v1.2.4-rancher6
+version: v1.2.4-rancher7
```
Comment (alena1108) [inline templates/kubernetes/config.yml]: during my initial review I thought we'd include @prachidamle changes for k8s, but we are postponing it to 1.2. So no new k8s image, and as the template version correlates to k8s image id (currently v1.2.4-rancher6), we should perhaps use v1.2.4-rancher6-1 or v1.2.4-rancher6.1 as per https://github.com/rancher/rancher-catalog/pull/163

@ibuildthecloud @vincent99 which format should we go for?

Comment (LLParse) [inline templates/kubernetes/config.yml]: Per slack conversation, we decided on `v1.2.4-rancher6.1`

**Comment by vincent99** [comment]
`v1.2.4-rancher6.1`

File: templates/kubernetes/2/docker-compose.yml [hunk 5]
```diff
@@ -44,47 +44,30 @@ proxy:
...
     io.rancher.container.start_once: 'true'
+    io.rancher.sidekicks: data
   environment:
     ETCD_DATA_DIR: /data
     ETCDCTL_ENDPOINT: http://etcd:2379
+    RANCHER_DEBUG: ${DEBUG}
   links:
-  - etcd-data
+  - data
   - discovery
```
Comment (alena1108) [inline templates/kubernetes/2/docker-compose.yml]: what does this link point to? there is no "discovery" service in the template

File: templates/kubernetes/2/docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+    io.rancher.scheduler.affinity:container_label_ne: io.rancher.stack_service.name=$${stack_name}/$${service_name}
+    io.rancher.container.start_once: 'true'
+    io.rancher.sidekicks: data
+  environment:
+    ETCD_DATA_DIR: /data
+    ETCDCTL_ENDPOINT: http://etcd:2379
+    RANCHER_DEBUG: ${DEBUG}
+  links:
+  - data
+  - discovery
```
Comment (alena1108) [inline templates/kubernetes/2/docker-compose.yml]: what does this link point to? there is no "discovery" service in the template

**Comment by alena1108** [comment]
LGTM

**Pull request merged**